### PR TITLE
doc: add system dependencies to README

### LINKED ISSUE #255: doc: Add system dependencies to README
@fzzzy noted a few missing libraries when trying to compile locally. Creating issue here to track them, and to later update README with the system dependency lists.

these include:
libssl-dev
libmysql-dev

### PR DESCRIPTION
**Pull request by amrita72**
## Description

Add system dependencies to README

## Issue(s)

Closes #255.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,18 +28,20 @@ Mozilla Sync Storage built with [Rust](https://rust-lang.org).
...
+- gcc
+- [golang](https://golang.org/doc/install)
+- libcurl4-openssl-dev
+- libmysqlclient-dev
+- libssl-dev
+- make
+- pkg-config
 - [Rust stable](https://rustup.rs)
-- MySQL 5.7 (or compatible)
-  -\* libmysqlclient (`brew install mysql` on macOS, `apt-get install libmysqlclient-dev` on Ubuntu)
```
Comment (jrconlin) [inline README.md]: I think we might want to leave the mac install instructions. We've got some folks who do dev work on macs and have asked how to install mysql.

Comment (amrita72) [reply README.md]: Sure, I'll revert this. Anything else you may want to add?

Comment (jrconlin) [reply README.md]: No, I think that's the only bit. Thanks for fixing up this document.

**Comment by amrita72** [comment]
@jrconlin @pjenvey I've fixed the commit message and description in this PR too.

**Pull request merged**